jgrapht-core/src/main/java/org/jgrapht/generate/GeneralizedPetersenGraphGenerator.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * (C) Copyright 2017-2017, by Joris Kinable and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.generate;
+
+import org.jgrapht.Graph;
+import org.jgrapht.Graphs;
+import org.jgrapht.VertexFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Generator for <a href="http://mathworld.wolfram.com/GeneralizedPetersenGraph.html">Generalized Petersen Graphs</a>
+ * The Generalized Petersen graphs $GP(n,k)$ are a family of cubic graphs formed by connecting the vertices of a
+ * regular polygon (cycle graph $C_n$) to the corresponding vertices of a star polygon ${n,k}$.
+ * Several special cases of the generalized Petersen graph are predefined in the {@link NamedGraphGenerator}.
+ *
+ * @author Joris Kinable
+ *
+ * @param <V> graph vertex type
+ * @param <E> graph edge type
+ */
+public class GeneralizedPetersenGraphGenerator<V, E>
+        implements GraphGenerator<V, E, V> {
+
+    private final int n;
+    private final int k;
+
+    /**
+     * Constructs a GeneralizedPetersenGraphGenerator used to generate a Generalized Petersen graphs $GP(n,k)$.
+     * @param n size of the regular polygon (cycle graph $C_n$)
+     * @param k size of the star polygon ${n,k}$
+     */
+    public GeneralizedPetersenGraphGenerator(int n, int k){
+        if(n<3)
+            throw new IllegalArgumentException("n must be larger or equal than 3");
+        if(k<1 || k>Math.floor((n-1)/2.0))
+            throw new IllegalArgumentException("k must be in the range [1, floor((n-1)/2.0)]");
+
+        this.n=n;
+        this.k=k;
+    }
+
+    @Override
+    public void generateGraph(Graph<V, E> target, VertexFactory<V> vertexFactory, Map<String, V> resultMap) {
+        List<V> verticesU=new ArrayList<>(n);
+        List<V> verticesV=new ArrayList<>(n);
+        for(int i=0; i<n; i++){
+            verticesU.add(vertexFactory.createVertex());
+            verticesV.add(vertexFactory.createVertex());
+        }
+        Graphs.addAllVertices(target, verticesU);
```

Comment:
@jsichi That would make sense, but is unfortunately not possible. The resultMap is of the type Map<String,V>. There are multiple vertices which should be labeled 'star' or 'polygon'. It would probably make more sense if the resultmap is of the type Map<V, String>.